### PR TITLE
Using SummaryCard instead of Table for image and Pod details

### PR DESCRIPTION
**Pull request by dabasnidhi**
1. Replaced table with summary card in pod and image details.
2. Handling Zero pods scenario for services.
Test-pod: http://nidabas-desk:8096/summary?namespace=default

### CODE REVIEW

File: src/WebUI/Common/KubeSummary.scss [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+        padding-top: 6px;
+        padding-bottom: 6px;
+        display: flex;
+    }
+
+    .details-card-info-field-size {
+        margin-right: 8px;
+        width: 150px;
+        min-width: 150px;
+    }
```
Comment (madhuv-msft) [inline src/WebUI/Common/KubeSummary.scss]: sorry to bother, as figma, 
min-width and width make it as 96px, margin-right make it as 20px; accordingly adjust the value field also.

also check the padding between title of the card and first row, as per figma it is 20px, means, 6px for padding and 14px for the otherone, same with the bottom padding also.

File: src/WebUI/Common/KubeSummary.scss [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
+
+    .details-card-value-field-size {
+        margin-right: 8px;
+        position: absolute;
+        width: calc(100% - 158px);
+        margin-left: 158px;
+    }
```
Comment (madhuv-msft) [inline src/WebUI/Common/KubeSummary.scss]: make margin-right as 20px, 
width: calc(100% - 116px)
margin-left is 116px

File: src/WebUI/ImageDetails/ImageDetails.tsx [hunk 3]
```diff
@@ -152,51 +123,47 @@ export class ImageDetails extends BaseComponent<IImageDetailsProperties, IImageD
...
-        const itemToRender = (
-            <Tooltip text={key} overflowOnly>
-                <span className={css("text-ellipsis")}>{key}</span>
-            </Tooltip>
+    private _getCardContent = (): JSX.Element => {
+        const items = this._getImageDetailsRowsData(this.props.imageDetails);
+        return (
+            <div
+                className="flex-column"
+                style={{ flexWrap: "wrap", width: "100%" }}
```
Comment (madhuv-msft) [inline src/WebUI/ImageDetails/ImageDetails.tsx]: make this as a css rather being here, use a classlike details-card-content, something like that

File: src/WebUI/ImageDetails/ImageDetails.tsx [hunk 4]
```diff
@@ -152,51 +123,47 @@ export class ImageDetails extends BaseComponent<IImageDetailsProperties, IImageD
...
+            >
+                {items.map((item, index) => (
+                    <div className="flex-row details-card-row-size bolt-list-cell" key={index}>
```
Comment (madhuv-msft) [inline src/WebUI/ImageDetails/ImageDetails.tsx]: bolt-list-cell --> do we need this css?

File: src/WebUI/Pods/PodOverview.tsx [hunk 5]
```diff
@@ -97,83 +89,57 @@ export class PodOverview extends BaseComponent<IPodOverviewProps> {
...
-                minWidth: 400,
-                renderCell: PodOverview._renderValueCell
-            }
-        ];
+    private _getCardContent = (): JSX.Element => {
+        const items = PodOverview._getPodDetails(this.props.pod, this.props.showImageDetails);
+        return (
+            <div
+                className="flex-column"
+                style={{ flexWrap: "wrap", width: "100%" }}
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodOverview.tsx]: same comment as above,

File: src/WebUI/Pods/PodsActionsCreator.ts [hunk 6]
```diff
@@ -18,16 +18,29 @@ export class PodsActionsCreator extends ActionCreatorBase {
...
+        if (fetchByLabel && !labelSelector) {
+            // For service's associated pods, fetchByLabel is true; In this scenario if no label selector is supplied, it implies zero pods
+            const podList: V1PodList = {
+                apiVersion: "",
+                items: [],
+                kind: "",
+                metadata: {} as V1ListMeta
+            };
+            // Calling action inside timeout to ensure the action listener is initialized in the ServiceDetails
+            setTimeout(() => this._actions.podsFetchedByLabel.invoke({ podsList: podList, labelSelector: "", isLoading: false }));
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodsActionsCreator.ts]: ideally this should not the case, in servicedetails, if we have the following code post getstore code, that will take care of this. just a thought

this._servicesStore.addListener(ServicesEvents.ServicePodsFetchedEvent, this._onPodsFetched);

Comment (dabasnidhi) [reply src/WebUI/Pods/PodsActionsCreator.ts]: I tried adding listener immediately after getStore, but that does not work. So calling action from timeout.

File: src/WebUI/ImageDetails/ImageDetails.tsx (hunk 4 above)
Comment (dabasnidhi) [reply src/WebUI/ImageDetails/ImageDetails.tsx]: Removed bolt-list-cell and added relative position to details-card-row-size

File: src/WebUI/ImageDetails/ImageDetails.tsx (hunk 3 above)
Comment (dabasnidhi) [reply src/WebUI/ImageDetails/ImageDetails.tsx]: Done

File: src/WebUI/Common/KubeSummary.scss (hunk 1 above)
Comment (dabasnidhi) [reply src/WebUI/Common/KubeSummary.scss]: Done. Have made changes to ensure the UI paddings/margins/fonts are same as in figma

File: src/WebUI/Common/KubeSummary.scss (hunk 2 above)
Comment (dabasnidhi) [reply src/WebUI/Common/KubeSummary.scss]: Done. Have made changes to ensure the UI paddings/margins/fonts are same as in figma

File: src/WebUI/Common/KubeSummary.scss [hunk 7]
```diff
@@ -1,15 +1,37 @@
...
\ No newline at end of file
+
+    .details-card-content {
+        flex-wrap: wrap;
+        width: 100%;
+
+        .details-card-row-size {
+            padding-top: 6px;
+            padding-bottom: 6px;
+            display: flex;
```
Comment (madhuv-msft) [inline src/WebUI/Common/KubeSummary.scss]: add to the div

File: src/WebUI/Common/KubeSummary.scss [hunk 8]
```diff
@@ -1,15 +1,37 @@
...
+        }
+
+        .details-card-info-field-size {
+            width: 96px;
+            min-width: 96px;
+        }
+
+        .details-card-value-field-size {
+            position: absolute;
+            width: calc(100% - 136px);
```
Comment (madhuv-msft) [inline src/WebUI/Common/KubeSummary.scss]: add a comment to say, label has left of 20px and we want to have 20px left of value, so total 40px + width of label. that is why 136px

File: src/WebUI/Common/KubeSummary.scss (hunk 7 above)
Comment (dabasnidhi) [reply src/WebUI/Common/KubeSummary.scss]: Removed display flex as it was not required

File: src/WebUI/Common/KubeSummary.scss (hunk 8 above)
Comment (dabasnidhi) [reply src/WebUI/Common/KubeSummary.scss]: Added comment for explaining 136px logic

**Pull request merged**